src/thorlabs_cube/driver/kcube/kpa.py

```diff
@@ -0,0 +1,483 @@
+        """
+        payload = st.pack(
+            "<HFFFFFFHH",
+            QUAD_POSITION_DEMAND_OUTPUTS_SUB_ID,
+            p_gain,
+            i_gain,
+            d_gain,
+            d_cutoff_freq,
+            notch_freq,
+            filter_q,
+            notch_on,
+            deriv_filter_on,
+        )
+        await self.send(Message(MGMSG.QUAD_SET_PARAMS, data=payload))
+
+    async def get_quad_loop_params2(
+        self,
+    ) -> tuple[float, float, float, float, float, float, int, int]:
+        """Get the extended loop parameters for the quad system.
+
+        :return: A tuple containing p_gain, i_gain, d_gain, d_cutoff_freq,
+        notch_freq, filter_q, notch_on, deriv_filter_on.
+        """
+        get_msg = await self.send_request(
+            MGMSG.QUAD_REQ_PARAMS, [MGMSG.QUAD_GET_PARAMS], _CHANNEL
+        )
+        return st.unpack("<HFFFFFFHH", get_msg.data[2:29])
+
+    async def set_trigger_config(
+        self,
+        trig1_mode: int,
+        trig1_polarity: int,
+        trig1_sum_min: int,
+        trig1_sum_max: int,
+        trig1_diff_threshold: int,
+        trig2_mode: int,
+        trig2_polarity: int,
+        trig2_sum_min: int,
+        trig2_sum_max: int,
+        trig2_diff_threshold: int,
+    ) -> None:
+        """Set trigger configuration for both TRIG1 and TRIG2.
+
+        :param trig1_mode: TRIG1 operating mode.
+        :param trig1_polarity: TRIG1 polarity.
+        :param trig1_sum_min: TRIG1 sum minimum.
+        :param trig1_sum_max: TRIG1 sum maximum.
+        :param trig1_diff_threshold: TRIG1 differential threshold.
+        :param trig2_mode: TRIG2 operating mode.
+        :param trig2_polarity: TRIG2 polarity.
+        :param trig2_sum_min: TRIG2 sum minimum.
+        :param trig2_sum_max: TRIG2 sum maximum.
+        :param trig2_diff_threshold: TRIG2 differential threshold.
+        """
+        payload = st.pack(
+            "<HHHHHHHHHHHHH",
+            trig1_mode,
+            trig1_polarity,
+            trig1_sum_min,
+            trig1_sum_max,
+            trig1_diff_threshold,
+            trig2_mode,
+            trig2_polarity,
+            trig2_sum_min,
+            trig2_sum_max,
+            trig2_diff_threshold,
+            _RESERVED,
+            _RESERVED,
+            _RESERVED,
+        )
+        await self.send(Message(MGMSG.QUAD_SET_PARAMS, data=payload))
+
+    async def get_trigger_config(
+        self,
+    ) -> tuple[int, int, int, int, int, int, int, int, int, int]:
+        """Get trigger configuration for both TRIG1 and TRIG2.
+
+        :return: A tuple containing trigger configuration parameters for TRIG1 and TRIG2.
+        """
+        get_msg = await self.send_request(
+            MGMSG.QUAD_REQ_PARAMS, [MGMSG.QUAD_GET_PARAMS], _CHANNEL
+        )
+        return st.unpack("<HHHHHHHHHHHHH", get_msg.data[2:27])
+
+    async def set_digital_outputs(self, dig_ops: int) -> None:
```

Comment:
Please name this argument something more informative. My first thought was that it meant 'digital operations'.